src/Views/IMEView.cpp

```diff
@@ -0,0 +1,130 @@
+#include <utility>
+
+#include <utility>
+
+#include "IMEView.h"
+
+
+IMEView::IMEView() {
+	log_printf(DBG_DEBUG, "IMEView::IMEView()");
+	auto sce_common_dialog_config_param = SceCommonDialogConfigParam{};
+	sceCommonDialogSetConfigParam(&sce_common_dialog_config_param);
+	me_ptr = std::shared_ptr<IMEView>(this);
+}
+
+
+void IMEView::openIMEView(IMEViewResult *result, std::string title, SceUInt32 maxInputLength) {
+	openIMEView(result, std::move(title), "", maxInputLength);
+}
+
+void IMEView::openIMEView(IMEViewResult *result, std::string title, std::string initialText, SceUInt32 maxInputLength) {
```

Comment:
Why is this method static if it's a singleton anyway? Making it not static would remove the need to use the `me_ptr` dirty hack.

Reply:
The `me_ptr` is to ensure, that at least one reference to the object exists throughout the program (so that it doesn't get destructed at some point). This wouldn't be necessary if the `Singleton` implementation would also use smart pointers.

Reply:
I should probably just have used global objects instead of the `Singleton` pattern, because its advantages are not obvious...

I've tried to add a global variable but it crashes on start, and I'm not sure exactly why.